src/Sarif/Core/SarifLog.cs

```diff
@@ -115,7 +116,46 @@ public void ApplyPolicies()
 
             foreach (Run run in this.Runs)
             {
-                run.ApplyPolicies();
+                run.ApplyPolicies(policiesCache);
+            }
+        }
+
+        /// <summary>
+        /// Applies the policies contained from a sarif file
+        /// </summary>
+        public void ApplyPolicies(string sarifLogPath)
+        {
+            SarifLog sarifLog = Load(sarifLogPath);
+            if (sarifLog == null || sarifLog.Runs == null)
```

Comment:
So this code appears to fall back to the internal policies on this log, if any, if the external file can't be loaded. Is that right? If someone specifies an external policy file that doesn't have anything, should we stop processing policy altogether?

That strikes me as cleaner: if you specify any external policy, it replaces use of any internal policies (on the log being processed).

Reply:
ok. Let me make that change!

---
In reply to: [517495418](https://github.com/microsoft/sarif-sdk/pull/2144#discussion_r517495418) [](ancestors = 517495418)